yellow-paper/docs/cryptography/performance-targets.md

```diff
@@ -0,0 +1,121 @@
+# Honk targets and win conditions
+
+## Introduction & context
+
+Aztec's cryptography tech stack and its associated implementation is an open-ended project with potential for many enhancements, optimisations and scope-creep.
+
+This document is designed to definitively answer the following questions:
+
+1. What are the metrics we care about when measuring our cryptography components?
+3. What are minimum satisfiable values for these metrics?
+4. What are the aspirational values for these metrics?
+
+# Important Metrics
+
+The following is a list of the relevant properties that affect the performance of the Aztec network:
+
+* Size of a user transaction (in kb)
+* Time to generate a user transaction proof
+* Memory required to generate a user transaction proof
+* Time to generate an Aztec Virtual Machine proof
+* Memory required to generate an Aztec Virtual Machine proof
+* Time to compute a 2x2 rollup proof
+* Memory required to compute a 2x2 rollup proof
+
+<!-- We can break these properties down into metrics linked to specitic cryptographic components:
+
+* Size of Goblin Plonk proofs
+* Size of Honk proofs
+* Honk prover time
+* Goblin Plonk prover time
+* Protogalaxy recursion -->
+
+ "MVP" = minimum standards that we can go to main-net with.
+
+| metric | how to measure | MVP | ideal |
+| --- | --- | --- | --- |
+| proof size | total size of a user tx incl. goblin plonk proofs | 32kb | 8kb |
+| prover time | 8 iterations of protogalaxy with 2^17 circuits (web browser) | 1 min | 10 seconds |
```

Comment:
Are these "8 iterations" 8 "app function calls"?
Or is this 4 "app function calls" and 4 "kernel iterations"?

Reply:
Follow-up Q's:
1. How many protogalaxy folds can be done?
2. We're planning on "breaking the kernel circuit up" into little circuits that do 1 thing (e.g. tree membership checks). Is it faster to fold lots of smaller instances or a few bigger instances?
3. If a single tx needs to execute 4 different "app functions" (circuits), and each of those circuits is a different size (i.e. each fits inside a different power of 2), is folding possible? Or can you only fold circuits which have matching 'row counts'?

Reply:
8 app function calls.

1: I think we are planning on supporting folding 2 instances into 1 accumulator (i.e. a kernel circuit + a function circuit). If more folds are required they need to be split up into repeated folds (e.g. 1 circuit can fold 2 instances into an accumulator, and then fold another 2 instances into the accumulator)
2: Faster to fold a few bigger instances. The number of constraints required by the folding verifier is mostly independent of the circuit sizes being folded
3: Every circuit being folded needs to have a defined maximum size (likely 2^19 due to memory constraints). In theory, folding a size-m circuit into a size-n accumulator (where m < n) can hav a Prover time that is mostly O(m). note: our current protogalaxy implementation does not do this but we can likely modify it to do so